### PR TITLE
[INLONG-8442][Manager] Support delete tenant and tenant role

### LINKED ISSUE #8442: [Feature][Manager] Support delete tenant and tenant role
### Description

inlong manager add tow interfaces:

- Delete tenant
- Delete tenant role

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8442   

### Motivation
inlong manager support  delete tenant and delete tenant role

### Modifications
The details can be found at [#8442   ](https://github.com/apache/inlong/issues/8442).

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java [hunk 1]
```diff
@@ -116,4 +116,9 @@ public TenantRoleInfo getByUsernameAndTenant(String name, String tenant) {
         return CommonBeanUtils.copyProperties(entity, TenantRoleInfo::new);
     }
 
+    @Override
+    public boolean delete(Integer id) {
+        return tenantUserRoleEntityMapper.deleteById(id) > 0;
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java:121]: No need to check if the delete result bigger than 0, just return it directly.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantController.java [hunk 2]
```diff
@@ -82,4 +83,13 @@ public Response<Boolean> update(@Validated(UpdateByIdValidation.class) @RequestB
         return Response.success(tenantService.update(request));
     }
 
+    @RequestMapping(value = "/tenant/delete/{name}", method = RequestMethod.GET)
+    @ApiOperation(value = "Delete inlong tenant by name")
+    @ApiImplicitParam(name = "name", dataTypeClass = String.class, required = true)
+    @RequiresRoles(logical = Logical.OR, value = {INLONG_ADMIN, INLONG_OPERATOR})
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantController.java:89]: INLONG_OPERATOR has no permission to do this.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java [hunk 3]
```diff
@@ -81,4 +81,11 @@ public Response<PageInfo<TenantRoleInfo>> listByCondition(@RequestBody TenantRol
         return Response.success(tenantRoleService.listByCondition(request));
     }
 
+    @RequestMapping(value = "/role/tenant/delete/{id}", method = RequestMethod.DELETE)
+    @ApiOperation(value = "Delete tenant role by ID")
+    @ApiImplicitParam(name = "id", dataTypeClass = Integer.class, required = true)
+    @RequiresRoles(logical = Logical.OR, value = {UserRoleCode.TENANT_ADMIN, UserRoleCode.TENANT_OPERATOR})
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java:87]: ditto

Comment (castorqin) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java]: done

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantController.java (hunk 2 above)
Comment (castorqin) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantController.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java (hunk 1 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java [hunk 4]
```diff
@@ -116,4 +116,16 @@ public TenantRoleInfo getByUsernameAndTenant(String name, String tenant) {
         return CommonBeanUtils.copyProperties(entity, TenantRoleInfo::new);
     }
 
+    @Override
+    public boolean delete(Integer id) {
+        int rowCount = tenantUserRoleEntityMapper.deleteById(id);
+        if (rowCount != InlongConstants.AFFECTED_ONE_ROW) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java:122]: `Preconditions.expectTrue(success == 1, "delete failed");`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java [hunk 5]
```diff
@@ -107,4 +107,11 @@ public Boolean update(InlongTenantRequest request) {
         }
         return true;
     }
+
+    @Override
+    public Boolean delete(String name, String operator) {
+        InlongTenantEntity inlongTenantEntity = inlongTenantEntityMapper.selectByName(name);
+        return inlongTenantEntityMapper.deleteById(inlongTenantEntity.getId()) > 0;
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:114]: Preconditions.expectTrue(success == 1, "delete failed");

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java [hunk 6]
```diff
@@ -52,4 +52,11 @@ public interface TenantRoleService {
      * Get one tenant role by name and tenant
      */
     TenantRoleInfo getByUsernameAndTenant(String name, String tenant);
+
+    /**
+     * Delete tenant role by id
+     * @param id tenant role
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:58]: Add one blank between doc description and param tag, thanks.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java [hunk 7]
```diff
@@ -52,4 +52,11 @@ public interface TenantRoleService {
...
+     * @return true = delete success/ false = delete fail
+     */
+    boolean delete(Integer id);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:61]: Change the return type to `Boolean` please.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java (hunk 4 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java (hunk 6 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:65]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java (hunk 7 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java (hunk 5 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java]: done

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java [hunk 8]
```diff
@@ -121,6 +122,14 @@ public Boolean update(InlongTenantRequest request) {
         return true;
     }
 
+    @Override
+    public Boolean delete(String name) {
+        InlongTenantEntity inlongTenantEntity = inlongTenantEntityMapper.selectByName(name);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:127]: Add some log. like `LOGGER.info("begin to delete inlong tenant name={} by user={}", name, operator);`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java [hunk 9]
```diff
@@ -121,6 +122,14 @@ public Boolean update(InlongTenantRequest request) {
         return true;
     }
 
+    @Override
+    public Boolean delete(String name) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:126]:
```suggestion
    public Boolean delete(String name, String operator) {
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java [hunk 10]
```diff
@@ -121,6 +122,14 @@ public Boolean update(InlongTenantRequest request) {
...
+        InlongTenantEntity inlongTenantEntity = inlongTenantEntityMapper.selectByName(name);
+        int success = inlongTenantEntityMapper.deleteById(inlongTenantEntity.getId());
+        Preconditions.expectTrue(success == 1, "delete failed");
+        return true;
```
[the remark this answers is not in the document]
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:132]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java [hunk 11]
```diff
@@ -126,4 +126,11 @@ public List<String> listTenantByUsername(String username) {
         return tenantUserRoleEntityMapper.listByUsername(username);
     }
 
+    @Override
+    public Boolean delete(Integer id) {
+        int success = tenantUserRoleEntityMapper.deleteById(id);
```
[the remark this answers is not in the document]
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java:134]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java (hunk 9 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:126]: done

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**